admiral/pkg/clusters/serviceentry.go

```diff
@@ -260,6 +262,17 @@ func modifyServiceEntryForNewServiceOrPod(
 			util.LogElapsedTimeSinceForModifySE(ctxLogger, "AdmiralCacheCreateServiceEntryForDeployment",
 				deploymentOrRolloutName, deploymentOrRolloutNS, rc.ClusterID, "", start)
 			modifySEerr = common.AppendError(modifySEerr, errCreateSE)
+
+			if common.IsVSBasedRoutingEnabled() {
```

Comment:
Coud we start introducing interface and decouple the interaction of service entry? This would help test units in isolation and not have integration style unit tests. we would be benefit from injecting mock implementations in the unit tests. 

WDYT?

Reply:
thats a valid point and it has been discussed several times in the past to refactor the modifySE code. Some parts of the code does employ interface driven mocking strategy but it is specifically lacking in this piece of code.

Reply:
Where are we adding the sni value in DestinationRules?

Reply:
@nirvanagit , with your fix in istio we would probably won't need to add that. So this implementation currently works with istio versions that contain your fix.
We could make it backwards compatible and I'll be doing that later in upcoming PRs.

Reply:
Got it.

So the plan is to keep the VirtualServices ready, and then test out by modifying Gateway to use PASSTHROUGH in a dev setup?

Reply:
@nirvanagit Yes we did test changing the to passthrough and rolling back to auto_passthrough, it works without errors. Once the VS is in place